operator/pkg/reconciliation/construct_statefulset.go

```diff
@@ -125,7 +126,15 @@ func newStatefulSetForCassandraDatacenterHelper(
 	}
 
 	// workaround for https://cloud.google.com/kubernetes-engine/docs/security-bulletins#may-31-2019
-	if dc.Spec.ServerType == "dse" {
+
+	serverImage := ""
+	for _, c := range template.Spec.Containers {
+		if c.Name == CassandraContainerName {
+			serverImage = c.Image
+		}
+	}
+
+	if dc.Spec.ServerType == "dse" || images.DockerImageRunsAsCassandra(dc.Spec.ServerVersion, serverImage) {
```

Comment:
What if a user explicitly specifies an image? How do we know if it uses the cassandra user or not? What if its image name doesn't follow the same regex?

Reply:
Good catch.  I just added a commit with a test case with an unparseable image name that defaults to assuming cassandra is running as root.

Reply:
My feeling is that we should default to the thing we want to be true (which I assume is use the Cassandra user). Like what if I'm testing some version of the cassandra image still in dev (where my tag might just be a git hash)? Do we need a flag to permit the user to make explicit whether or not the cassandra user is in use to override inferring from the image string?

Reply:
I suggested a flag like that at one point.

How about:

1. If we don't recognize the image name, we default to assuming the image runs as cassandra
2. We look for CASSOP_RUN_AS_CASSANDRA="false" to override the default?

@jimdickinson ^ ?

Reply:
Ok, after discussing this in standup, I have provided a DockerImageRunsAsCassandra boolean in the Datacenter spec.  It is optional and can be left undefined.  If it is defined and set to true, then we will define the security context.  

Otherwise we fall back to the calculation logic in images.CalculateDockerImageRunsAsCassandra()

Note that CalculateDockerImageRunsAsCassandra() will default to true if it cannot recognize the image and version combination, which will cause the securitycontext to be defined.